### PR TITLE
Adds initial Spec of slider web component

### PR DESCRIPTION
**Pull request by procload**
This is a draft PR for the spec of the slider component.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/0b8e596e512b17f42d382be384d9c0df9c43cc0b'>0b8e596e512b17f42d382be384d9c0df9c43cc0b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by procload** [comment]
Moving this back to "Draft" status as I found the SliderLabel FAST Foundation component when looking through the FAST docs. This should drastically simplify many of the outstanding questions I had in the Slider Spec. I think it makes sense to handle these as two separate components or PRs, however. I think we can implement the Slider as-is w/out the label and hold off on publishing it until the SliderLabel is finished as well. I don't expect a lot of work for the SliderLabel, but will create a Spec for it as well.

**Comment by procload** [comment]
@miroslavstastny and @chrisdholt I know you're both swamped, but this spec is ready for review and questions whenever you get a chance.

File: packages/web-components/src/slider/slider.spec.md
Comment (miroslavstastny) [inline packages/web-components/src/slider/slider.spec.md:23]: FUIR9 only supports single value. The range option is not in the Figma spec.

File: packages/web-components/src/slider/slider.spec.md
Comment (miroslavstastny) [inline packages/web-components/src/slider/slider.spec.md:39]: `track` slot is named `rail` in both FUIR9 and Figma

File: packages/web-components/src/slider/slider.spec.md
Comment (miroslavstastny) [inline packages/web-components/src/slider/slider.spec.md:84]: If I understand the Figma spec correctly, the `range` there means the value indicators before and after the slider.

**Review by miroslavstastny — COMMENTED** [review]
LGTM, the only thing to discuss it the `rail` slot in FUIR9 vs `track` in Fast.
Figma spec uses both.

File: packages/web-components/src/slider/slider.spec.md
Comment (chrisdholt) [reply packages/web-components/src/slider/slider.spec.md:39]: if we have good data from Open UI, we can propose updating this

File: packages/web-components/src/slider/slider.spec.md
Comment (procload) [reply packages/web-components/src/slider/slider.spec.md:84]: Not that they wouldn't need to implement the labels, just that the SliderLabel is a separate component. Could probably word this better.

File: packages/web-components/src/slider/slider.spec.md
Comment (chrisdholt) [reply packages/web-components/src/slider/slider.spec.md:23]: If it's there now but wasn't before, let's (for now) take is a follow up given I'm not sure that range exists in FAST at this time either.

File: packages/web-components/src/slider/slider.spec.md
Comment (chrisdholt) [inline packages/web-components/src/slider/slider.spec.md:35]: Are we forking the FAST template and slotting elements as a pattern for how form elements work? This feels more like a construct from the react implementation where what is being returned is an input...but you can "slot" your own if you want. This won't work the same w/ custom elements and this is not how the FAST Form Associated elements work or will work.

File: packages/web-components/src/slider/slider.spec.md
Comment (chrisdholt) [inline packages/web-components/src/slider/slider.spec.md:38]: If range doesn't exist in v9 and we aren't picking up now, we should remove these.

File: packages/web-components/src/slider/slider.spec.md
Comment (chrisdholt) [inline packages/web-components/src/slider/slider.spec.md:50]: Will this be baked in? The implications of this are that slider now has a dependency on tooltip _in all scenarios_.

File: packages/web-components/src/slider/slider.spec.md
Comment (chrisdholt) [inline packages/web-components/src/slider/slider.spec.md:82]: `value` is what sets the default value. I believe that defaultValue is a react construct for inputs. There's a long issue  here which illustrates the entire evolution of why we are keeping `value` as initial/default value instead of changing how it works: https://github.com/microsoft/fast/issues/5119

Comment (chrisdholt) [reply packages/web-components/src/slider/slider.spec.md:82]: TLDR, if we change how `value` works in an HTML custom element, it's unlikely that the platform will change to align with us - doing so would "break the internet". In this case, we elected to add a new attribute which has a better chance of making it into the platform over time than changing how value works.

**Review by chrisdholt — CHANGES_REQUESTED** [review]
I think there are a few concepts here where we need to understand deltas. For instance, if the default slot takes an input, that's making assumptions about how users interact with this and how form associated custom elements work.

File: packages/web-components/src/slider/slider.spec.md
Comment (procload) [reply packages/web-components/src/slider/slider.spec.md:23]: I'm good with kicking this down the road until the design spec is more fleshed out, but it looks like it does exist in [the FAST Slider attributes](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/slider/slider.spec.md#api).

File: packages/web-components/src/slider/slider.spec.md
Comment (procload) [reply packages/web-components/src/slider/slider.spec.md:35]: Gotcha, makes sense. Thanks.

File: packages/web-components/src/slider/slider.spec.md
Comment (procload) [reply packages/web-components/src/slider/slider.spec.md:50]: I don't think we should bake it in. Sounds like you agree. Let's leave it out and I'll update the Spec to document the differences between this and React.

File: packages/web-components/src/slider/slider.spec.md
Comment (procload) [reply packages/web-components/src/slider/slider.spec.md:39]: @chrisdholt What's the typical process you have used to determine the data from Open UI on things like this?

File: packages/web-components/src/slider/slider.spec.md
Comment (procload) [reply packages/web-components/src/slider/slider.spec.md:82]: I need more clarity here. If we go ahead with the `current-value` attribute from FAST, is it just a matter of reflecting those deviations from Fluent React in the Spec?

File: packages/web-components/src/slider/slider.spec.md
Comment (procload) [reply packages/web-components/src/slider/slider.spec.md]: Updated it to reflect that the default slot reflects child elements of the component, such as the SliderLabel component.

File: packages/web-components/src/slider/slider.spec.md
Comment (chrisdholt) [reply packages/web-components/src/slider/slider.spec.md]: I guess my thought here is - we need to decide if this is an area worth forking in order to maintain compatability with react. `defaultValue` in my understanding here is a _react_ construct for inputs - the custom elements leverage Form Association and so there is no construct of "defaultValue". On the HTML platform, `value` is literally the default/initial value...it doesn't two way bind, it doesn't update, etc. It is an "input" to the input which gives a default value from the view for a given form element. That's just how the platform works. IMO this is an earned difference and we can document as such. FAST in this case is aligned to how the web platform works today and how it will work moving forward (I don't see a world where an input's value attribute changes in behavior, again this will "break the internet"). This is a _react_ deviation from the platform to enable certain features. I'm not sure how we want to treat this, but from a standards perspective, the deviation here isn't around how value works.

Comment (procload) [reply packages/web-components/src/slider/slider.spec.md]: I see the arguments for sticking with the web platform, but I don't see the arguments for maintaining compatibility with React, especially if it's going against the grain. @miroslavstastny do you have any strong opinions on maintaining compatibility with React on this, or are you OK with documenting the differences?

**Comment by chrisdholt** [comment]
> LGTM, the only thing to discuss it the `rail` slot in FUIR9 vs `track` in Fast. Figma spec uses both.

+ @procload - @miroslavstastny if you can share the audit or research via Open UI we can raise this to align in FAST.

File: packages/web-components/src/slider/slider.spec.md
Comment (miroslavstastny) [reply packages/web-components/src/slider/slider.spec.md:36]: OK, it seems Open UI prefers track: https://open-ui.org/components/slider.research.parts#part-concepts
I suggest to stick with `track` and document it as an earned difference

File: packages/web-components/src/slider/slider.spec.md
Comment (miroslavstastny) [reply packages/web-components/src/slider/slider.spec.md]: The `defaultValue` is React-specific concept to support a default value for an [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html).
If you pass `value` prop to a React component, it becomes controlled - the component itself cannot change the `value`, it is controlled from the parent. The `defaultValue` allows as to define a default/initial value while keeping the component as uncontrolled.

As this is a React specific concept, we should not follow the pattern in web components and do whatever is natural there.

**Comment by miroslavstastny** [comment]
> > LGTM, the only thing to discuss it the `rail` slot in FUIR9 vs `track` in Fast. Figma spec uses both.
> 
> * @procload - @miroslavstastny if you can share the audit or research via Open UI we can raise this to align in FAST.

Let's stick with `track` it seems to be preferred https://github.com/microsoft/fluentui/pull/26830#discussion_r1114203539

**Comment by procload** [comment]
Thanks, guys! I have a decent amount of the CSS and Storybook done. I'll need to switch gears to the SliderLabel component next. I also have a PR out in FAST Foundation to swap the direction of the vertical orientation slider.

**Comment by chrisdholt** [comment]
@procload I think most things are resolved here but it looks like the branch is out of sync. If we can get that updated/fixed I'll unblock and we can have @miroslavstastny do a final pass and then I think we're good to go ahead and merge this.

**Comment by micahgodbolt** [comment]
This PR seems to have way more than just the Slider spec. Is that intentional?

**Comment by procload** [comment]
> This PR seems to have way more than just the Slider spec. Is that intentional?

No, it wasn't. Working through some merge/rebase workflow issues. I'll likely revert the commits here to only include the Spec, or close this out and re-open with the spec. Apologies.

**Comment by procload** [comment]
I had inadvertently introduced a merge/rebase error in this PR. This PR was only supposed to include the Slider.spec.md file and the empty folder. I'm closing it and [re-opened it here](https://github.com/microsoft/fluentui/pull/26981). Apologies for the Github churn.

**Pull request closed without merging**
